api/src/main/java/org/apache/iceberg/actions/ActionsProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.Table;
+
+/**
+ * An API that should be implemented by query engine integrations for providing actions.
+ */
+public interface ActionsProvider {
+
+  /**
+   * Instantiates an action to snapshot an existing table.
+   */
+  default SnapshotTable snapshotTable(String sourceTableIdent, String destTableIdent) {
```

Comment:
One of the issues with the old API is that always assumed it operated on a single existing table. This method was offered as a static method there, which looked weird. In the new API, it is no longer a static method.

Reply:
It's too bad we can't have a required method. I'd much rather not rely on order of arguments here and pass the destination name separately:

```
SparkActions.get().snapshotTable("a.b").as("a.c").execute();
```

Reply:
I think your concern is valid. Supporting `snapshotTable("db.tbl").as("db.tbl_2")` isn't ideal as we don't pass the required param right away but I'll be okay throwing an exception if the destination name is not set by calling `as`.
That will be a bit better than what we have now.

Reply:
Well, we could also default it to something like `db.tbl_iceberg_snapshot`.